### PR TITLE
Remove rpaths that point outside the prefix

### LINKED ISSUE #457: Remove rpaths that point outside of the prefix
conda-build removes rpath entries that point to e.g. the "build" prefix

```py
    for old in existing:
        if old.startswith("$ORIGIN"):
            new.append(old)
        elif old.startswith("/"):
            # Test if this absolute path is outside of prefix. That is fatal.
            rp = relpath(old, prefix)
            if rp.startswith(".." + os.sep):
                print(f"Warning: rpath {old} is outside prefix {prefix} (removing it)")
            else:
                rp = "$ORIGIN/" + relpath(old, origin)
                if rp not in new:
                    new.append(rp)
```

### PR DESCRIPTION
**Pull request by orhun**
closes #457

See https://github.com/conda/conda-build/blob/03f2a2af949abc2d5a7a20d86ac1d06af0b7bc17/conda_build/post.py#L622

e.g.

```
WARN rattler_build::linux::link: rpath ("/home/orhun/gh/rattler-build/output/bld/rattler-build_zlink_1704804443/build_env/lib") is outside of prefix ("/home/orhun/gh/rattler-build/output/bld/rattler-build_zlink_1704804443/host_env_placehold_.....") - removing it
```

### CODE REVIEW

File: src/linux/link.rs [hunk 1]
```diff
@@ -100,15 +100,23 @@ impl SharedObject {
...
+                    &new_rpath,
                     self.path.parent().ok_or(RelinkError::NoParentDir)?,
                 )
                 .ok_or(RelinkError::PathDiffFailed)?;
-                tracing::info!("New relative path: $ORIGIN/{}", relative_path.display());
-                final_rpath.push(PathBuf::from(format!(
-                    "$ORIGIN/{}",
-                    relative_path.to_string_lossy()
-                )));
+                if relative_path.starts_with("..") {
```
Comment (wolfv) [inline src/linux/link.rs:107]: Does `Component::Parent` work with `starts_with`, too? For Path's.

Comment (orhun) [reply src/linux/link.rs:107]: Yes,

```rust
use std::path::Path;

fn main()  {
    let path = Path::new("../test");
    assert!(path.starts_with(".."));
    assert_eq!(Some(Path::new("..")), path.parent());
}
```

Do you prefer checking that instead?

**Comment by baszalmstra** [comment]
Im wondering if we should also emit which file contains the rparh?

**Review by wolfv — APPROVED** [review]

**Comment by orhun** [comment]
> Im wondering if we should also emit which file contains the rparh?
can you elaborate? @baszalmstra

**Comment by baszalmstra** [comment]
If I understand it correctly a warning is emited if a binary file is found that contains an rpath outside of the build prefix. 

But the warning does not specify which file contained the rpath if Im not mistaken. I was thinking that that would be useful information.

**Comment by orhun** [comment]
Ah I see what you mean, done in b98613dec574b7713c8954ba0130ad195c125eea

**Review by wolfv — APPROVED** [review]

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**